opts/opts.go

```diff
@@ -515,3 +516,26 @@ func (m *MemSwapBytes) UnmarshalJSON(s []byte) error {
 	b := MemBytes(*m)
 	return b.UnmarshalJSON(s)
 }
+
+type OomScoreAdj int64
+
+func (o *OomScoreAdj) Type() string { return "int64" }
+
+func (o *OomScoreAdj) Value() int64 { return int64(*o) }
+
+func (o *OomScoreAdj) String() string {
+
+	val := strconv.FormatInt(int64(*o), 10)
+	return val
+}
+
+func (o *OomScoreAdj) Set(value string) error {
+
+	var conv int64
+	conv, err := strconv.ParseInt(value, 10, 64)
+	if err != nil || conv < -1000 || conv > 1000 {
+		return err
+	}
+	*o = OomScoreAdj(conv)
+	return nil
+}
```

Comment:
Bit on the fence here if we should have a special option-type for this. It does allow for validation on the client-side, but also makes assumption about what the daemon-side allows (given; it's unlikely to change); for `docker run` it looks like we leave the validation to the API, and using a regular `IntVar` option; https://github.com/docker/cli/blob/64206aef767d23420803badbd205d1c5a661f097/cli/command/container/opts.go#L306